src/osdc/Objecter.cc

```diff
@@ -2056,11 +2056,11 @@ ceph_tid_t Objecter::_op_submit_with_budget(Op *op, RWLock::Context& lc, int *ct
 
   // throttle.  before we look at any state, because
   // _take_op_budget() may drop our lock while it blocks.
-  if (!op->ctx_budgeted || (ctx_budget && (*ctx_budget == -1))) {
```

Comment:
the first part of this condition is usually true - (!op->ctx_budgeted), and we intentionally want to budget for most ops in this case. The updated version never calls _take_op_budget() is these cases, which would be a bug.